workitem/workitem_repository.go

```diff
@@ -690,48 +693,173 @@ func (r *GormWorkItemRepository) Fetch(ctx context.Context, spaceID uuid.UUID, c
+				ELSE concat(path::text, '.', REPLACE(id::text, '-', '_'))::ltree
+			END AS pathself,
+			id
+	FROM %s)
+	SELECT array_agg(iterations.id)::text AS children,
+		PathResolver.id::text AS iterationid
+	FROM %s,
+		PathResolver
+	WHERE path <@ PathResolver.pathself
+	AND space_id = ?
+	GROUP BY (PathResolver.pathself,
+		PathResolver.id)`,
+		iterationTableName,
+		iterationTableName)
+	db = r.db.Raw(queryIterationWithChildren, spaceID.String())
+	db.Scan(&itrChildren)
+	if db.Error != nil {
+		log.Error(ctx, map[string]interface{}{
+			"space_id": spaceID.String(),
+			"err":      db.Error,
+		}, "unable to fetch children for every iteration in a space")
+		return nil, errors.NewInternalError(db.Error.Error())
+	}
+	childMap := map[string][]string{}
+	for _, r := range itrChildren {
+		// Following can be done by implementing Valuer interface for type IterationHavingChildrenID
+		r.Children = strings.TrimPrefix(r.Children, "{")
+		r.Children = strings.TrimRight(r.Children, "}")
+		children := strings.Split(r.Children, ",")
+		childMap[r.IterationID] = children
+	}
 	countsMap := map[string]WICountsPerIteration{}
-	for _, iterationWithCount := range res {
-		countsMap[iterationWithCount.IterationID] = iterationWithCount
+	for _, i := range wiMap {
+		t := i.Total
+		c := i.Closed
+		if children, exists := childMap[i.IterationID]; exists {
+			for _, child := range children {
+				if _, exists := wiMap[child]; exists {
+					t += wiMap[child].Total
+					c += wiMap[child].Closed
+				}
+			}
+		}
+		countsMap[i.IterationID] = WICountsPerIteration{
+			IterationID: i.IterationID,
+			Total:       t,
+			Closed:      c,
+		}
 	}
 	return countsMap, nil
 }
 
 // GetCountsForIteration returns Closed and Total counts of WI for given iteration
 // It executes
 // SELECT count(*) as Total, count( case fields->>'system.state' when 'closed' then '1' else null end ) as Closed FROM "work_items" where fields@> concat('{"system.iteration": "%s"}')::jsonb and work_items.deleted_at is null
-func (r *GormWorkItemRepository) GetCountsForIteration(ctx context.Context, iterationID uuid.UUID) (map[string]WICountsPerIteration, error) {
+func (r *GormWorkItemRepository) GetCountsForIteration(ctx context.Context, itr *iteration.Iteration) (map[string]WICountsPerIteration, error) {
 	var res WICountsPerIteration
-	query := fmt.Sprintf(`SELECT count(*) as Total,
-						count( case fields->>'system.state' when 'closed' then '1' else null end ) as Closed
-						FROM "work_items"
-						where fields@> concat('{"system.iteration": "%s"}')::jsonb
-						and work_items.deleted_at is null`, iterationID)
-	db := r.db.Raw(query)
+	pathOfIteration := append(itr.Path, itr.ID)
+	// get child IDs of the iteration
+	var childIDs []uuid.UUID
+	iterationTable := iteration.Iteration{}
+	iterationTableName := iterationTable.TableName()
+	getIterationsOfSpace := fmt.Sprintf(`SELECT id FROM %s WHERE path <@ ? and space_id = ?`, iterationTableName)
```

Comment:
@kwk I reverted back because I was facing the same error for table name
`SELECT id FROM ""  WHERE (path <@ '46dc5ab8_ad6d_418f_a3d5_8b0716aa7a97' and space_id = '2e0698d8-753e-4cef-bb7c-f027634824a2')`
This was the quiry being built.